### PR TITLE
tiles: new firedancer port support in xdp, net tile

### PR DESCRIPTION
**Pull request by lheeger-jump**
Adds support for gossip and repair ports in XDP and fdctl config. Also adds support for ports to be unset in the config.

### CODE REVIEW

File: src/app/fdctl/configure/xdp.c [hunk 1]
```diff
@@ -80,12 +80,34 @@ init( config_t * const config ) {
...
+    config->tiles.repair.repair_intake_listen_port,
+    config->tiles.repair.repair_serve_listen_port,
+  };
+
+  ushort udp_ports[ NUM_UDP_PORT_CANDIDATES ];
+  ulong udp_ports_cnt = 0;
+
+  /* if at this point one of these udp listen ports is 0 it was not 
+     configured. */
+  for( ulong i = 0; i<NUM_UDP_PORT_CANDIDATES; i++ ) {
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/xdp.c:99]:
```suggestion
  for( ulong i = 0UL; i<NUM_UDP_PORT_CANDIDATES; i++ ) {
```

File: src/app/fdctl/configure/xdp.c [hunk 2]
```diff
@@ -80,12 +80,34 @@ init( config_t * const config ) {
...
+    config->tiles.quic.regular_transaction_listen_port,
+    config->tiles.quic.quic_transaction_listen_port,
+    config->tiles.shred.shred_listen_port,
+    config->tiles.gossip.gossip_listen_port,
+    config->tiles.repair.repair_intake_listen_port,
+    config->tiles.repair.repair_serve_listen_port,
+  };
+
+  ushort udp_ports[ NUM_UDP_PORT_CANDIDATES ];
+  ulong udp_ports_cnt = 0;
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/xdp.c:95]:
```suggestion
  ulong  udp_ports_cnt = 0;
```

File: src/app/fdctl/configure/xdp.c [hunk 3]
```diff
@@ -80,12 +80,34 @@ init( config_t * const config ) {
...
+
+  /* if at this point one of these udp listen ports is 0 it was not 
+     configured. */
+  for( ulong i = 0; i<NUM_UDP_PORT_CANDIDATES; i++ ) {
+    if( udp_port_candidates[ i ]==0 ) {
```
Comment (mmcgee-jump) [inline src/app/fdctl/configure/xdp.c:100]: if( FD_UNLIKELY( !...) ) )

File: src/app/fdctl/configure/xdp.c (hunk 1 above)
Comment (lheeger-jump) [reply src/app/fdctl/configure/xdp.c]: done

File: src/app/fdctl/configure/xdp.c (hunk 2 above)
Comment (lheeger-jump) [reply src/app/fdctl/configure/xdp.c]: done

File: src/app/fdctl/configure/xdp.c (hunk 3 above)
Comment (lheeger-jump) [reply src/app/fdctl/configure/xdp.c]: done

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**